<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpticVariationWidget</class>
 <widget class="QWidget" name="OpticVariationWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>821</width>
    <height>733</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Optic variation method (orbit flattening)</string>
  </property>
  <layout class="QGridLayout" columnstretch="1,1">
   <item row="0" column="0" colspan="2">
    <layout class="QHBoxLayout" name="_2" stretch="0,1,0,0,0,0,0,0">
     <item>
      <widget class="QLabel">
       <property name="text">
        <string>Configuration:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="configComboBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="editConfigButton">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel">
       <property name="text">
        <string>Mode:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="modeXButton">
       <property name="text">
        <string>x</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttongroup_xy</string>
       </attribute>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="modeYButton">
       <property name="text">
        <string>y</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttongroup_xy</string>
       </attribute>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="modeXYButton">
       <property name="text">
        <string>x/y</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttongroup_xy</string>
       </attribute>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="1">
    <widget class="QGroupBox">
     <property name="title">
      <string>Steerer corrections</string>
     </property>
     <layout class="QVBoxLayout" name="gridLayout_2">
      <item>
       <layout class="QHBoxLayout">
        <item>
         <widget class="QCheckBox" name="backtrackCheckBox">
          <property name="text">
           <string>Backtrack</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel">
          <property name="text">
           <string>Method:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="methodMatchButton">
          <property name="text">
           <string>MATCH</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttongroup_method</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="methodORMButton">
          <property name="text">
           <string>ORM</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttongroup_method</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="methodSectormapButton">
          <property name="text">
           <string>sectormap</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttongroup_method</string>
          </attribute>
         </widget>
        </item>
        <item>
         <spacer>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="TableView" name="resultsTable">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QPushButton" name="applyButton">
          <property name="text">
           <string>Apply</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="prevButton">
          <property name="text">
           <string>...</string>
          </property>
          <property name="arrowType">
           <enum>Qt::LeftArrow</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="nextButton">
          <property name="text">
           <string>...</string>
          </property>
          <property name="arrowType">
           <enum>Qt::RightArrow</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Automatic</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0">
        <widget class="QPlainTextEdit" name="logEdit">
         <property name="tabChangesFocus">
          <bool>true</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="startProcedureButton">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="abortProcedureButton">
           <property name="text">
            <string>Cancel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <widget class="QProgressBar" name="progressBar">
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel">
           <property name="text">
            <string>Ignored shots:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="numIgnoredSpinBox">
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel">
           <property name="text">
            <string>Used shots:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="numUsedSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="manualTabWidget">
      <attribute name="title">
       <string>Manual</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QComboBox" name="opticComboBox"/>
         </item>
         <item>
          <widget class="QPushButton" name="setOpticButton">
           <property name="text">
            <string>Set optic</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0">
          <widget class="QPushButton" name="updateButton">
           <property name="text">
            <string>Update</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="fitButton">
           <property name="text">
            <string>Fit</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="recordButton">
           <property name="text">
            <string>Record</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox">
     <property name="title">
      <string>Records</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <item row="1" column="0">
       <widget class="QPushButton" name="clearRecordsButton">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="removeRecordsButton">
        <property name="text">
         <string>Remove</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="exportRecordsButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Export</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <widget class="TableView" name="recordsTable">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox">
     <property name="title">
      <string>Targets</string>
     </property>
     <layout class="QVBoxLayout">
      <item>
       <widget class="TableView" name="targetsTable">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox">
     <property name="title">
      <string>Optics</string>
     </property>
     <layout class="QGridLayout" columnstretch="0,0,0">
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel">
          <property name="text">
           <string>Init from focus:</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QComboBox" name="read1stFocusComboBox"/>
        </item>
        <item>
         <widget class="QLabel">
          <property name="text">
           <string>/</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="read2ndFocusComboBox"/>
        </item>
       </layout>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="readFocusButton">
        <property name="text">
         <string>Read</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="TableView" name="opticsTable"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox">
     <property name="title">
      <string>Monitor readouts</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="TableView" name="monitorTable"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TableView</class>
   <extends>QTableView</extends>
   <header>madgui/widget/tableview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttongroup_method"/>
  <buttongroup name="buttongroup_xy"/>
 </buttongroups>
</ui>
